keps/sig-windows/2258-node-service-log-viewer/README.md

```diff
@@ -0,0 +1,373 @@
+available [here](https://github.com/kubernetes/kubernetes/pull/96120).
+
+#### kubectl
+
+`kubectl` has an existing `logs` command that is used to view the logs for a
+container in a pod or a specified resource. The sub-command looks at resource
+types, so can be extended to work with node objects to view the logs of services
+on the nodes. Given that the `logs` command depends on RBAC policies for access
+to appropriate resource type and associated endpoints, it will allow us to
+restrict node logs access to only cluster administrators as long as the cluster
+is setup in that manner.
+
+The default mode for `logs` sub-command for node objects would be to query
+the systemd journal on supported Linux nodes and Application, System, and ETW
+on Windows nodes, allowing searching, time and service based filtering. A
+`--path` argument will be added to see a list of log files available under 
+`/var/logs/` or `C:\var\logs\` and view those contents directly.
+
+
+Here are some examples and explanation of the options that will be added.
+```
+Examples:
+  # Show kubelet logs from all masters
+  kubectl logs nodes --role master -s kubelet
+
+  # See what logs are available in masters in /var/logs
+  kubectl logs nodes --role master --path=/
+
+  # Show docker logs from Windows nodes
+  kubectl logs nodes -l kubernetes.io/os=windows --path=journal -s docker
+
+  # Show logs from all applications logging to the event logs on Windows nodes
+  kubectl logs nodes -l kubernetes.io/os=windows --path=journal
+
+Options:
+      --case-sensitive=true: Filters are case sensitive by default. Pass --case-sensitive=false to do a case insensitive filter.
+  -g, --grep='': Filter log entries by the provided regex pattern. Only applies to node journal logs.
+  -o, --output='': Display journal logs in an alternate format (short, cat, json, short-unix). Only applies to node journal logs.
+      --path='journal': Retrieve the specified path within the node's /var/logs/ folder. The 'journal' value will allow querying the journal or Get-WinEvent on supported operating systems.
+      --raw=false: Perform no transformation of the returned data.
+      --role='': Set a label selector by node role.
+  -l, --selector='': Selector (label query) to filter on.
+      --since='': Return logs after a specific ISO timestamp or relative date. Only applies to node journal or Get-WinEvent logs.
+      --tail=0: Return up to this many lines (not more than 100k) from the end of the log. Only applies to node journal or Get-WinEvent logs.
+      --unify=false: Interleave logs by sorting the output. Defaults on when viewing node journal logs.
+  -s, --service=[]: Return log entries from the specified service(s). Only applies to node journal or Get-WinEvents logs.
+      --until='': Return logs before a specific ISO timestamp or relative date. Only applies to node journal logs.
+```
+
+Given that the feature will be introduced behind a feature gate, by default
```

Comment:
we need to add a security section... about not breaking out of the /var/log/path

Reply:
`/var/log` is an existing `kubelet` endpoint that is not being introduced as part of this KEP. We are only extending it.